pallets/pallet-migration/src/lib.rs

```diff
@@ -0,0 +1,249 @@
+		>,
+	{
+		#[pallet::call_index(0)]
+		#[pallet::weight({
+			//TODO: Placeholder.
+			Weight::from_parts(1_000_000, 1_000_000)
+		})]
+		pub fn update_balance(origin: OriginFor<T>, requested_migrations: EntriesToMigrate<T>) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			requested_migrations.attestation.iter().try_for_each(|key| {
+				let is_migrated = Self::is_key_migrated(Attestations::<T>::hashed_key_for(key))?;
+				if !is_migrated {
+					attestation::migrations::update_balance_for_entry::<T>(key)
+				} else {
+					Ok(())
+				}
+			})?;
+
+			requested_migrations.delegation.iter().try_for_each(|key| {
+				let is_migrated = Self::is_key_migrated(DelegationNodes::<T>::hashed_key_for(key))?;
+				if !is_migrated {
+					delegation::migrations::update_balance_for_entry::<T>(key)
+				} else {
+					Ok(())
+				}
+			})?;
+
+			requested_migrations.did.iter().try_for_each(|key| {
+				let is_migrated = Self::is_key_migrated(Did::<T>::hashed_key_for(key))?;
+				if !is_migrated {
+					did::migrations::update_balance_for_entry::<T>(key)
+				} else {
+					Ok(())
+				}
+			})?;
+
+			requested_migrations.lookup.iter().try_for_each(|key| {
+				let is_migrated = Self::is_key_migrated(ConnectedDids::<T>::hashed_key_for(key))?;
+				if !is_migrated {
+					pallet_did_lookup::migrations::update_balance_for_entry::<T>(key)
+				} else {
+					Ok(())
+				}
+			})?;
+
+			requested_migrations.w3n.iter().try_for_each(|key| {
+				let is_migrated = Self::is_key_migrated(Owner::<T>::hashed_key_for(key))?;
+				if !is_migrated {
+					pallet_web3_names::migrations::update_balance_for_entry::<T>(key)
+				} else {
+					Ok(())
+				}
+			})?;
+
+			requested_migrations
+				.public_credentials
+				.iter()
+				.try_for_each(|(key, key2)| {
+					let is_migrated = Self::is_key_migrated(Credentials::<T>::hashed_key_for(key, key2))?;
+					if !is_migrated {
+						public_credentials::migrations::update_balance_for_entry::<T>(key, key2)
+					} else {
+						Ok(())
+					}
+				})?;
+
+			Self::deposit_event(Event::EntriesUpdated(requested_migrations));
+
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		fn is_key_migrated(key: Vec<u8>) -> Result<bool, DispatchError> {
+			let key_hash = <T as frame_system::Config>::Hashing::hash(&key[..]);
+
+			if MigratedKeys::<T>::contains_key(key_hash) {
+				return Ok(true);
+			}
+			MigratedKeys::<T>::insert(key_hash, ());
+			Ok(false)
+		}
+	}
+
+	impl<T: Config> MigrationManager<AccountIdOf<T>, BalanceOf<T>> for Pallet<T> {
+		fn exclude_key_from_migration(key: Vec<u8>) -> Result<(), DispatchError> {
+			let key_hash = <T as frame_system::Config>::Hashing::hash(&key[..]);
+			MigratedKeys::<T>::insert(key_hash, ());
+			Ok(())
+		}
+
+		fn is_key_migrated(key: Vec<u8>) -> Result<bool, DispatchError> {
+			let key_hash = <T as frame_system::Config>::Hashing::hash(&key[..]);
+			Ok(MigratedKeys::<T>::contains_key(key_hash))
+		}
```

Comment:
Why can this error?

Reply:
Looks like an artifact. removed in: [bd65957](https://github.com/KILTprotocol/kilt-node/pull/540/commits/bd65957b601e9cdc18dad2a8cf6e6a47a43d7e8c)